### PR TITLE
OpenJPEG update 2.5.2 - Add JPH read support.

### PR DESCRIPTION
**Pull request by blowekamp**
## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by blowekamp** [comment]
> @blowekamp good work. I suggest following that approach to first re-import the same old version we already have, and then run a normal update to the desired version as a second step. It can still all go in this MR.
> 
> Meanwhile I've temporarily disabled ghostflow's root commit rejection.

The previous version was 1.2.0 from 2007. The library had a different layout that is currently does. I am unsure if there is much advantage to tracking the files. Will the commit in the subtree automatically track the moving files? And how would the merges work with the changing locations of files?

**Comment by bradking** [comment]
Ah, okay.  In the tiff case our old import had diverged from its upstream version's layout so re-importing the old version to match the layout made sense.  That doesn't make sense here.

**Comment by bradking** [comment]
Meanwhile let's ask ghostflow to re-check this.

Do: check

**Comment by bradking** [comment]
@blowekamp I updated your branch to rebuild the import using a simpler approach:

1. Remove all of the old imported sources
2. Add the import script
3. Run the import script.  It works automatically now.
4. Add the local build system edits

The ghostflow checks are still not happy though.  I'm looking into them.

**Comment by bradking** [comment]
@blowekamp I updated your branch again to resolve the ghostflow check errors.  Now only the errors on the WIP commit are left.  You should be able to edit and force-push the last two commits (those after the merge commit) as appropriate.  I'll hand this back over to you now.

File: Modules/ThirdParty/OpenJPEG/src/openjpeg/src/lib/openjp2/CMakeLists.txt [hunk 1]
```diff
@@ -171,7 +190,10 @@ endif()
 #################################################################################
 set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
 
+#[[ -- ITK
 option(OPJ_USE_THREAD "Build with thread/mutex support " ON)
+# -- ITK]]
+set(OPJ_USE_THREAD OFF)
```
Comment (dzenanz) [inline Modules/ThirdParty/OpenJPEG/src/openjpeg/src/lib/openjp2/CMakeLists.txt:209]: Why do we turn this off?

**Review by dzenanz — APPROVED** [review]
Looks good

File: Modules/ThirdParty/OpenJPEG/src/openjpeg/src/lib/openjp2/CMakeLists.txt (hunk 1 above)
Comment (blowekamp) [reply Modules/ThirdParty/OpenJPEG/src/openjpeg/src/lib/openjp2/CMakeLists.txt:209]: This appear to be an option to support multithreaded decoding. Our ImageIO does not use this feature, and didn't think it was work the effort to investigate further.

Comment (dzenanz) [reply Modules/ThirdParty/OpenJPEG/src/openjpeg/src/lib/openjp2/CMakeLists.txt:209]: It should be easy to flip or expose if we change our mind later.

**Comment by hjmjohnson** [comment]
@blowekamp Failures seem unrelated to this PR.  Perhaps there was a networking issue where downloads failed during builds?

**Comment by hjmjohnson** [comment]
/azp.run ITK.Linux

**Comment by hjmjohnson** [comment]
/azp.run ITK.Linux.Python

**Comment by hjmjohnson** [comment]
\azp.run ITK.Linux.Python

**Comment by blowekamp** [comment]
I manually tested the JPEG200ImageIO with some test files from the OpenJPH collection:
https://github.com/aous72/jp2k_test_codestreams

The reading worked well. I was hoping to be able to create a jph file with some of these tools, but they seem only able to create the code stream and not the jph file structure.

**Pull request merged**